src/poprox_storage/repositories/accounts.py

```diff
@@ -56,44 +56,38 @@ def fetch_accounts_between(self, start_date, end_date) -> list[Account]:
 
         return self._fetch_acounts(query)
 
-    def fetch_account_by_email(self, email: str) -> Account | None:
+    def fetch_account_by_email_query(self, email_query: str) -> list[Account]:
         account_tbl = self.tables["accounts"]
         query = sqlalchemy.select(
-            account_tbl.c.account_id,
-            account_tbl.c.email,
-            account_tbl.c.status,
-            account_tbl.c.source,
-            account_tbl.c.created_at,
-        ).where(account_tbl.c.email == email)
-        result = self.conn.execute(query).fetchall()
-        accounts = [
-            Account(
-                account_id=row.account_id,
-                email=row.email,
-                status=row.status,
-                source=row.source,
-                created_at=row.created_at,
-            )
-            for row in result
-        ]
+            account_tbl,
+        ).where(account_tbl.c.email.like(email_query))
+        return self._fetch_acounts(query)
+
+    def fetch_account_by_email(self, email: str) -> Account | None:
+        account_tbl = self.tables["accounts"]
+        query = sqlalchemy.select(account_tbl).where(account_tbl.c.email == email)
+        accounts = self._fetch_acounts(query)
+
         if len(accounts) > 0:
             return accounts[0]
         return None
 
-    def store_account(self, account: Account) -> UUID | None:
+    def store_account(self, account: Account, commit=True) -> UUID | None:
         account_tbl = self.tables["accounts"]
-        query = (
-            sqlalchemy.insert(account_tbl)
-            .values(
-                account_id=account.account_id,
-                email=account.email,
-                source=account.source,
-                status="new_account",
-            )
-            .returning(account_tbl.c.account_id)
+        return self._upsert_and_return_id(
+            self.conn,
+            account_tbl,
+            {
+                "account_id": account.account_id,
+                "email": account.email,
+                "source": account.source,
+                "subsource": account.subsource,
+                "status": account.status,
+            },
+            # NOTE -- this is not explicitly named in out migrations and therefore is _fragile_
+            constraint="account_pkey",
+            commit=commit,  # do not commit so we throw errors.
```

Comment:
This didn't commit before so it seems like this changes the behavior. Maybe the default should be `commit=False`?